vault/resource_consul_secret_backend.go

```diff
@@ -296,13 +296,17 @@ func consulSecretsBackendCustomizeDiff(_ context.Context, diff *schema.ResourceD
 	newToken := diff.Get("token").(string)
 	newBootstrap := diff.Get("bootstrap").(bool)
 
-	// If the user sets bootstrap to false but doesn't provide a token, disallow it.
-	if newToken == "" && !newBootstrap {
+	// Disallow the following:
+	//   1. Bootstrap is false, the token field is empty, and we know this is the final value of token.
+	if !newBootstrap && newToken == "" && diff.NewValueKnown("token") {
```

Comment:
I like to group related booleans:

```suggestion
	if !newBootstrap && (newToken == "" && diff.NewValueKnown("token")) {
```